lib/wallet/src/Cardano/Wallet/DB.hs

```diff
@@ -332,6 +343,308 @@ data DBLayer m s k = forall stm. (MonadIO stm, MonadFail stm) => DBLayer
+    { dbWallets :: DBWallets stm s
+    , dbCheckpoints :: DBCheckpoints stm s
+    , dbWalletMeta :: DBWalletMeta stm
+    , dbDelegation :: DBDelegation stm
+    , dbTxHistory :: DBTxHistory stm
+    , dbPendingTxs :: DBPendingTxs stm
+    , dbPrivateKey :: DBPrivateKey stm k
+
+    -- The following two functions will need to be split up
+    -- and distributed the smaller layer parts as well.
+    , rollbackTo_
+        :: WalletId
+        -> Slot
+        -> ExceptT ErrNoSuchWallet stm ChainPoint
+    , prune_
+        :: WalletId
+        -> Quantity "block" Word32
+        -> ExceptT ErrNoSuchWallet stm ()
+    , atomically_
+        :: forall a. stm a -> m a
+    }
+
+-- | Create a legacy 'DBLayer' from smaller database layers.
+mkDBLayerFromParts
+    :: (MonadIO stm, MonadFail stm)
+    => DBLayerCollection stm m s k -> DBLayer m s k
+mkDBLayerFromParts DBLayerCollection{..} = DBLayer
+    { initializeWallet = initializeWallet_ dbWallets
+    , removeWallet = removeWallet_ dbWallets
+    , listWallets = listWallets_ dbWallets
+    , walletsDB = walletsDB_ dbCheckpoints
+    , putCheckpoint = putCheckpoint_ dbCheckpoints
+    , readCheckpoint = readCheckpoint_ dbCheckpoints
+    , listCheckpoints = listCheckpoints_ dbCheckpoints
+    , putWalletMeta = putWalletMeta_ dbWalletMeta
+    , readWalletMeta = readWalletMeta_ dbWalletMeta
+    , isStakeKeyRegistered = isStakeKeyRegistered_ dbDelegation
+    , putDelegationCertificate = putDelegationCertificate_ dbDelegation
+    , putDelegationRewardBalance = putDelegationRewardBalance_ dbDelegation
+    , readDelegationRewardBalance = readDelegationRewardBalance_ dbDelegation
+    , putTxHistory = putTxHistory_ dbTxHistory
+    , readTxHistory = readTxHistory_ dbTxHistory
+    , getTx = getTx_ dbTxHistory
+    , putLocalTxSubmission = putLocalTxSubmission_ dbPendingTxs
+    , readLocalTxSubmissionPending = readLocalTxSubmissionPending_ dbPendingTxs
+    , updatePendingTxForExpiry = updatePendingTxForExpiry_ dbPendingTxs
+    , removePendingOrExpiredTx = removePendingOrExpiredTx_ dbPendingTxs
+    , putPrivateKey = putPrivateKey_ dbPrivateKey
+    , readPrivateKey = readPrivateKey_ dbPrivateKey
+    , readGenesisParameters = readGenesisParameters_ dbWallets
+    , rollbackTo = rollbackTo_
+    , prune = prune_
+    , atomically = atomically_
+    }
+
+-- | A database layer for a collection of wallets
+data DBWallets stm s = DBWallets
+    { initializeWallet_
+        :: WalletId
+        -> Wallet s
+        -> WalletMetadata
+        -> [(Tx, TxMeta)]
+        -> GenesisParameters
+        -> ExceptT ErrWalletAlreadyExists stm ()
+        -- ^ Initialize a database entry for a given wallet. 'putCheckpoint',
+        -- 'putWalletMeta', 'putTxHistory' or 'putProtocolParameters' will
+        -- actually all fail if they are called _first_ on a wallet.
+
+    , readGenesisParameters_
+        :: WalletId
+        -> stm (Maybe GenesisParameters)
+        -- ^ Read the *Byron* genesis parameters.
```

Comment:
How about 
```suggestion
    , readByronGenesisParameters_
        :: WalletId
        -> stm (Maybe GenesisParameters)
```
?
(Let the code speak for itself)

Reply:
Well, it's only the comment here that mentions `Byron`, the definition of `GenesisParameters` does not mention Byron at all:

https://github.com/input-output-hk/cardano-wallet/blob/80756f919d7811219e23632bbe59214e0f5e25f5/lib/wallet/src/Cardano/Wallet/Primitive/Types.hs#L952-L963

Mainnet starts in the Byron, but in principle, the `GenesisParameters` may also apply to networks that start in a different era (`preview`?). In that case, the suggested function name becomes incorrect.

For this pull request, I would like to not touch the established nomenclature. ("If it was deemed appropriate before, then it will still be appropriate after merging.")